syntax = "proto3";

package routercommon;

// Domain for routing decision (geosite)
message Domain {
  enum Type {
    Plain = 0;
    Regex = 1;
    RootDomain = 2;
    Full = 3;
  }
  Type type = 1;
  string value = 2;
}

// IP for routing decision (geoip)
message CIDR {
  bytes ip = 1;
  uint32 prefix = 2;
}

message GeoIP {
  string country_code = 1;
  repeated CIDR cidr = 2;
}

message GeoIPList {
  repeated GeoIP entry = 1;
}

message GeoSite {
  string country_code = 1;
  repeated Domain domain = 2;
}

message GeoSiteList {
  repeated GeoSite entry = 1;
}
